flamel.py

```diff
@@ -127,15 +129,24 @@ def main():
         u_nks = parser.get_u_nks()
 
     # Step 2: Uncorrelate the data
-    if uncorrelator.needs_dhdls:
-        uncorrelator.set_dhdls(dhdls)
-    if uncorrelator.needs_u_nks:
-        uncorrelator.set_u_nks(u_nks)
-
-    if do_dhdl:
-        dhdls = uncorrelator.uncorrelate(dhdls, args.equiltime)
-    if do_u_nks:
-        u_nks = uncorrelator.uncorrelate(u_nks, args.equiltime)
+    if args.uncorr_threshold > 0:
+        if uncorrelator.needs_dhdls:
+            uncorrelator.set_dhdls(dhdls)
+        if uncorrelator.needs_u_nks:
+            uncorrelator.set_u_nks(u_nks)
+
+        if do_dhdl:
+            print("Uncorrelating dH/dl ...")
+            dhdls = uncorrelator.uncorrelate(dhdls, args.equiltime)
+        if do_u_nks:
+            print("Uncorrelating reduced potentials ...")
+            u_nks = uncorrelator.uncorrelate(u_nks, args.equiltime)
+
+    # concat data for estimators
```

Comment:
I moved the concatenation out of the correlation analysis functions because a threshold of 0 needs to skip correlators but still concatenates for estimators.